src/Symfony/Component/Messenger/Transport/Serialization/SerializerConfiguration.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Serialization;
+
+use Symfony\Component\Messenger\EnvelopeItem;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @experimental in 4.1
+ */
+final class SerializerConfiguration implements EnvelopeItem
+{
+    private $context;
+
+    public function __construct(array $context)
+    {
+        $this->context = $context;
+    }
+
+    public function getContext(): array
+    {
+        return $this->context;
+    }
+
+    public function serialize()
+    {
+        return serialize(array('context' => $this->context));
+    }
+
+    public function unserialize($serialized)
+    {
+        list('context' => $context) = unserialize($serialized, array('allowed_classes' => false));
+
+        $this->__construct($context);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getName(): string
+    {
+        return 'serializer_config';
```

Comment:
I think that the FQCN of the "item" is enough, right? We've already had these with the Form and went back to use the class, it's simpler (and names don't really bring much value)

Reply:
Fair enough.

Reply:
And also we can be calm about unique names :)